shimmingtoolbox/optimizer/quadprog_optimizer.py

```diff
@@ -0,0 +1,399 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import logging
+import quadprog
+
+from typing import List
+
+
+from shimmingtoolbox.optimizer.basic_optimizer import Optimizer
+from shimmingtoolbox.pmu import PmuResp
+from shimmingtoolbox.coils.coil import Coil
+
+ListCoil = List[Coil]
+logger = logging.getLogger(__name__)
+
+
+class QuadProgOpt(Optimizer):
+    """ Optimizer object that stores coil profiles and optimizes an unshimmed volume given a mask.
+        Use optimize(args) to optimize a given mask. The algorithm uses a least squares solver to find the best shim.
+        It supports bounds for each channel as well as a bound for the absolute sum of the channels.
+
+        Attributes:
+            _initial_guess_method (string) : String indicating how to find the first guess for the optimization
+            initial_coefs (np.ndarray): Initial guess that will be used in the optimization
+            reg_vector (np.ndarray) : Vector used to make the regulatization in the optimization
+            ineq_matrix (np.ndarray) : Matrix used in the quadratic optimization to apply the constraints
+            ineq_vector (np.ndarray) : Vector used in the quadratic optimization to represent the constraints
+    """
+
+    def __init__(self, coils: ListCoil, unshimmed, affine, reg_factor=0):
+        """
+        Initializes coils according to input list of Coil
+
+        Args:
+            coils (ListCoil): List of Coil objects containing the coil profiles and related constraints
+            unshimmed (numpy.ndarray): 3d array of unshimmed volume
+            affine (numpy.ndarray): 4x4 array containing the affine transformation for the unshimmed array
+            reg_factor (float): Regularization factor for the current when optimizing. A higher coefficient will
+                                penalize higher current values while a lower factor will lower the effect of the
+                                regularization. A negative value will favour high currents (not preferred).
+        """
+        super().__init__(coils, unshimmed, affine)
+        self._initial_guess_method = 'mean'
```

Comment:
```suggestion
        self.initial_guess_method = 'mean'
```

Also change this in the LS optimizer